HomeWorkAssigment/src/Main.java

```diff
@@ -0,0 +1,281 @@
+import java.util.*;
+
+public class Main {
+   
+        // Списки
+
+   /*1. Есть два класса: Address с полями улица и номер дома и
+        Person с полями имя и Address. Нужно написать функцию:
+
+        List<Address> getAddresses(List<Person> persons)
+
+        то есть по списку persons возвращать список их адресов. */
+        public static class Address {
+            String street;
+            int num;
+
+            public Address(String street, int num) {
+                this.street = street;
+                this.num = num;
+
+            }
+
+            @Override
+            public String toString() {
+                return "Address{" +
+                        "street='" + street + '\'' +
+                        ", num=" + num +
+                        '}';
+            }
+        }
+
+        public static class Person {
+            String name;
+            String Address;
+
+            public Person(String name, String Address) {
+                this.name = name;
+                this.Address = Address;
+            }
+
+            public List<String> getAddress(List<Person> persons) {
+                List<String> list = new LinkedList<>();
+                for (Person p : persons) {
+                    list.add(p.Address);
+                }
+                return list;
+            }
+        }
+
+   /*2. Есть список с именами: ​ Ivan, Maria, Stephan, John, Amalia​ .
+        Написать функцию, которая вернет список, в котором не содержатся имена исходного списка, длиной 4. */
+
+        public List<String> NamesWithoutFour () {
+            List<String> list = new ArrayList<>();
+            Iterator<String> iterator = list.iterator();
+
+            while (iterator.hasNext()) {
+                String name = iterator.next();
+                if (name.length() == 4) {
+                    iterator.remove();
```

Comment:
This is not bad, but you modify an original list, which may not be a good idea. It is alway better to create a new list and populate it with needed values (in this case, copy everything, but those with the length==4).